### PR TITLE
【BIT】转换能力补齐：paddle.nn.functional.[avg_pool1d avg_pool2d avg_pool3d]

### PR DESCRIPTION
**Pull request by cangtianhuang**
### 1. [paddle.nn.functional.avg_pool1d](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/avg_pool1d_cn.html#avg-pool1d)

查询 [torch.nn.functional.avg_pool1d 与 paddle.nn.functional.avg_pool1d 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.avg_pool1d.html) ，已有详细对比，发现是 `参数用法不一致` 。实际检查中发现，paddle 的
`padding` 参数更丰富，同时支持 TensorFlow 风格的 VALID 和 SAME，以及 PyTorch 风格的 int、list、tuple 类型。因此为其注册 AvgPoolRule 类。

由于在 torch.nn.functional.avg_pool1d 中仅支持对称填充，对于非对称填充，我们需要首先计算需要的填充大小，调用 torch.nn.functional.pad 手动进行填充，然后再调用 torch.nn.functional.avg_pool1d 方法进行计算。具体操作见 Rule 代码。

### 2. [paddle.nn.functional.avg_pool2d](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/avg_pool2d_cn.html#avg-pool2d)

与 paddle.nn.functional.avg_pool1d 类似，但更为复杂。查询 [torch.nn.functional.avg_pool2d 与 paddle.nn.functional.avg_pool2d 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.avg_pool2d.html) ，已有详细对比，发现是 `paddle 参数更多` 。同样为其注册 AvgPoolRule 类。

在本 api 中，paddle 的 `padding` 参数更加丰富，除了上述类型外，还有非对称的 4 个整数值、或 4 个二元组的形式（详情请查看 API 文档），难以重用代码，只能再次撰写手动处理和填充逻辑。具体操作见 Rule 代码。

paddle 相较于 torch 多了 `data_format` 参数，表示输入和输出的数据格式，可以是 "NCHW" 和 "NHWC"，而 torch 默认是 "NCHW"，因此需首先使用 permute 方法调整维度顺序，计算完毕后再使用 permute 调整回来。具体操作见 Rule 代码。

### 3. [paddle.nn.functional.avg_pool3d](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/avg_pool3d_cn.html#avg-pool3d)

与 paddle.nn.functional.avg_pool2d 类似，但更为复杂。查询 [torch.nn.functional.avg_pool3d 与 paddle.nn.functional.avg_pool3d 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.avg_pool3d.html) ，已有详细对比，发现是 `paddle 参数更多` 。同样为其注册 AvgPoolRule 类。

在本 api 中，paddle 的 `padding` 参数更加丰富，仍多了 `data_format` 参数，可以是"NCDHW"和"NDHWC"。再次撰写手动处理和填充逻辑。具体操作见 Rule 代码。

## 转换引擎增强

1. 为 mapping.json 中增加 `set_defaults`、`import` 字段
2. 调整 apply_generic 方法，使其返回 head 与 map 两段 code，更便于撰写 Rule 使用。

## 测试结果

相关测试配置全部通过，已全部移入 api_config_support2torch_1.txt（在 https://github.com/PFCCLab/PaddleAPITest/pull/78 中被移除，此 PR 中移进）

image

### CODE REVIEW

**Review by Cutelemon6 — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
